### PR TITLE
Beginning of separating out crates

### PR DESCRIPTION
**Pull request by benthecarman**
Separates out backend logic to `harbor-client` and the UI to `harbor-ui`. 

For follow ups will clean up deps and try to make the the separation more clean

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -20,4 +20,4 @@ clippy:
     cargo clippy --all-features --tests -- -D warnings
 
 reset-db:
-    diesel migration revert --all --database-url=harbor.sqlite && diesel migration run --database-url=harbor.sqlite
+    cd harbor-client && diesel migration revert --all --database-url=harbor.sqlite && diesel migration run --database-url=harbor.sqlite
```
Comment (oleonardolima) [inline justfile:23]: @benthecarman I've been wondering about this command, is it right as the default used is `~/.harbor` ?

**Review by oleonardolima — COMMENTED** [review]
tACK 32c924596ed10d15117df205b09aa30871af323a

I've successfully joined MutinyNet federation, received (via Lightning), and paid (via Onchain).

It compiles and builds successfully when using the Nix environment, otherwise it fails locally due to `sqlcipher` library, however the same error occurs on `master`.

File: justfile (hunk 1 above)
Comment (futurepaul) [reply justfile:23]: Yeah but looks like this was incorrect even before this pr. Will make an issue, good catch.

**Review by futurepaul — APPROVED** [review]
lgtm! will make an issue for @oleonardolima's comment

**Pull request merged**